doc/nrf/ug_bt_mesh_node_removal.rst

```diff
@@ -0,0 +1,198 @@
+To ensure that the removed node is not able to decrypt network messages after being reset, update the network key of the subnet the removed node belonged to.
+
+This procedure can be used in any other cases when a network key needs to be updated.
+For example, in cases where there is a suspicion of network credentials being compromised.
+The same procedure can be used to update an application key.
+
+.. note::
+   Updating keys is currently not supported by the nRF Mesh mobile app.
+
+The procedure of updating a network (or application) key is described below.
+
+Uploading a new key
+===================
+
+Upload a new network key (or application key) to the nodes by sending the Config NetKey (AppKey) Update message to every node, including the node with the Configuration Client model.
+
+The snippet below demonstrates how to do this using the |NCS| API:
+
+.. code-block:: c
+
+    static uint8_t send_net_key_update(struct bt_mesh_cdb_node *node, void *user_data)
+    {
+        struct bt_mesh_cdb_subnet *subnet = user_data;
+        uint8_t status = 0;
+        int err;
+
+        err = bt_mesh_cfg_cli_net_key_update(node->net_idx, node->addr, subnet->net_idx,
+                                             subnet->keys[1].net_key, &status);
+        if (err || status) {
+            printk("Unable to update NetKey on %#x (err %d, status %u)\n", node->addr, err,
+                   status);
+        } else {
+            printk("Updated NetKey with index %u on %#x\n", subnet->net_idx, node->addr);
+        }
+
+        return BT_MESH_CDB_ITER_CONTINUE;
+    }
+
+    static int update_net_key(uint16_t net_idx, uint8_t *new_net_key)
+    {
+        struct bt_mesh_cdb_subnet *subnet;
+
+        subnet = bt_mesh_cdb_subnet_get(net_idx);
+        if (!subnet) {
+            printk("Unable to get subnet\n");
+            return -EINVAL;
+        }
+
+        /* Store the new network key in CDB. */
+        memcpy(subnet->keys[1].net_key, new_net_key, 16);
+        bt_mesh_cdb_subnet_store(subnet);
+
+        /* Send the new network key to each node. */
+        bt_mesh_cdb_node_foreach(send_net_key_update, subnet);
+        return 0;
+    }
+
+    ...
+
+    uint8_t new_net_key[16] = { ... };
+    update_net_key(BT_MESH_NET_PRIMARY, new_net_key);
+
+Switching to the new key
+========================
+
+After uploading the new key, configure the nodes to use it when sending messages.
+To do that, change the Key Refresh Phase to 2 by sending the Config Key Refresh Phase Set message with Transition field set to 0x2.
+
+Send the message to at least one node.
+The rest of the nodes will switch the Key Refresh Phase through Secure Network Beacon.
+But in this case, it may take a while for all nodes to eventually switch the phase since the time between 2 consecutive Secure Network Beacons is ~10 seconds.
```

Comment:
```suggestion
But in this case, it may take a while for all nodes to eventually switch the phase since the time between 2 consecutive Secure Network Beacons is ~10 seconds and some nodes can backoff from sending secure network beacons up to 600 seconds.
```